components/scream/src/diagnostics/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+set(DIAGNOSTIC_SRCS
+  potential_temperature.cpp
+)
+
+set(DIAGNOSTIC_HEADERS
+  potential_temperature.hpp
+) 
+
+add_library(diagnostics ${DIAGNOSTIC_SRCS})
+target_include_directories(diagnostics PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../share)
+target_link_libraries(diagnostics scream_share)
```

Comment:
It's best to link libraries with `PUBLIC`, so that flags, include dirs, compile options, link options, etc. are forwarded to downstream targets.